### ISSUE TITLE
Removing wxRESIZE_BORDER window style flag has no effect on GTK

### ISSUE THREAD

**Issue by Journeyman-dev** [wxWidgets/wxWidgets#22767]
**Describe the bug**
When disabling the resize handles of a window in Fedora, the window is still resizable.

**Expected vs observed behavior**
Please describe what you expected to happen and what actually happens. Make sure to explain what exactly is the problem, just attaching a screenshot is not always sufficient. When in doubt, please provide more details rather than too few.

**Patch or snippet allowing to reproduce the problem**
Make a wxFrame inheriting class constructor like this:
```cpp
MyFrame::MyFrame()
    : wxFrame(nullptr, wxID_ANY, "My Frame")
{
  this->SetWindowStyle(wxDEFAULT_FRAME_STYLE & ~(wxRESIZE_BORDER));
}
```
The created window is still resizable by dragging the window sides with the mouse.

This issue is present in my app's release, which you can download here:
https://github.com/Journeyman-dev/FossSweeper/releases/tag/v0.1.0-beta
You can see the code in question similar to the snippet here:
https://github.com/Journeyman-dev/FossSweeper/blob/0f297160a6c20b274fe479c40bb9f554ee423ef5/modules/desktop_view/src/GameFrame.cpp#L54

**To Reproduce**
On platforms in question:
1. Create a wxFrame.
2. call `SetWindowStyle(wxDEFAULT_FRAME_STYLE & ~(wxRESIZE_BORDER))` on the frame.

**Platform and version information**
 - wxWidgets version you use: 3.3
 - wxWidgets port you use: wxGTK
 - OS and its version: Fedora 36
 - For wxGTK only 
   + GTK version: [e.g. 3.24.5]
   + Which GDK backend is used: X11
   + If relevant, i.e. for appearance-related problems: Adwaita

**Comment by vadz** [comment]
Just to make sure I understand correctly: do things work correctly when creating the frame initially without `wxRESIZE_BORDER` or is the frame resizeable even then?

**Comment by Journeyman-dev** [comment]
> Just to make sure I understand correctly: do things work correctly when creating the frame initially without `wxRESIZE_BORDER` or is the frame resizeable even then?

I just tried it now. It does not matter if the wxRESIZE_BORDER flag is present or not, the window is always resizable.

**Comment by vadz** [comment]
I can't reproduce under Gnome 42 with this diff, the frame is not resizeable as expected:
```diff
diff --git a/samples/minimal/minimal.cpp b/samples/minimal/minimal.cpp
index 470e765423..1d844419cf 100644
--- a/samples/minimal/minimal.cpp
+++ b/samples/minimal/minimal.cpp
@@ -139,7 +139,9 @@ bool MyApp::OnInit()

 // frame constructor
 MyFrame::MyFrame(const wxString& title)
-       : wxFrame(NULL, wxID_ANY, title)
+       : wxFrame(NULL, wxID_ANY, title,
+                 wxDefaultPosition, wxDefaultSize,
+                 wxDEFAULT_FRAME_STYLE & ~wxRESIZE_BORDER)
 {
     // set the frame icon
     SetIcon(wxICON(sample));
```

Does this "work" (i.e. doesn't work as expected, allowing to reproduce the bug) for you?

**Comment by Journeyman-dev** [comment]
Yes, doing it that way fixes the problem.

**Comment by Journeyman-dev** [comment]
https://github.com/wxWidgets/wxWidgets/issues/22767 is a related issue, seems your fix works for this too.

**Comment by oneeyeman1** [comment]
@Journeyman-dev ,
Not all style can be changed on the fly (after the window is created).

This is documented. But the list of such styles is not.

Thank you.

**Comment by vadz** [comment]
> Yes, doing it that way fixes the problem.

So you're saying that only resetting the flag later doesn't work (reconfirming because this contradicts your previous commit from just above)?

If so, yes, turning `wxRESIZE_BORDER` on or off after creation is not supported in wxGTK currently. It probably should be possible to fix this but this looks like a low priority issue to me as you usually know whether you want the window to be resizeable or not when creating it (and definitely a much lower priority one than if it didn't work at all).

**Comment by Journeyman-dev** [comment]
> So you're saying that only resetting the flag later doesn't work (reconfirming because this contradicts your previous commit from just above)?

Yes, this is what I am saying. I will do what you just said to fix my problem.

**Comment by mcastel1** [comment]
> I can't reproduce under Gnome 42 with this diff, the frame is not resizeable as expected:
> 
> ```diff
> diff --git a/samples/minimal/minimal.cpp b/samples/minimal/minimal.cpp
> index 470e765423..1d844419cf 100644
> --- a/samples/minimal/minimal.cpp
> +++ b/samples/minimal/minimal.cpp
> @@ -139,7 +139,9 @@ bool MyApp::OnInit()
> 
>  // frame constructor
>  MyFrame::MyFrame(const wxString& title)
> -       : wxFrame(NULL, wxID_ANY, title)
> +       : wxFrame(NULL, wxID_ANY, title,
> +                 wxDefaultPosition, wxDefaultSize,
> +                 wxDEFAULT_FRAME_STYLE & ~wxRESIZE_BORDER)
>  {
>      // set the frame icon
>      SetIcon(wxICON(sample));
> ```
> 
> Does this "work" (i.e. doesn't work as expected, allowing to reproduce the bug) for you?

I am using wxWidgets 3.2.5. This solution works on Windows 11, but not on OSX 14.6. 
On OSX 14.6 the wxFrame is still resizable.

**Comment by vadz** [comment]
@mcastel1 This issue is about wxGTK, so please open a different one about wxOSX as it's quite unrelated to everything else here.
